ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/TransactionHelper.java

```diff
@@ -26,16 +26,43 @@
 import javax.transaction.xa.XAResource;
 
 /**
+ * Utility class to make transaction status checking and resources registration easier.
+ *
  * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
  */
 public interface TransactionHelper {
 
+    /**
+     * Check if transaction is active. If error occurs wrap an original exception with {@link JMSException}.
+     *
+     * @return whether transaction is active or not.
+     * @throws JMSException if error occurred getting transaction status.
```

Comment:
the JMSException is thrown in case when some unexpected failure occurs in transaction manager, when SystemException is thrown from there: https://docs.oracle.com/javaee/7/api/javax/transaction/TransactionManager.html#getStatus--

I changed the sentence to: `if transaction service has failed in unexpected way to obtain transaction status`

I think the information should be put to each occurence of `@throws JMSException` rather than to the javadoc of the whole class. When I want to use some method want to see info directly at the method not to search somewhere at class level. Especially this is true when using standard IDE which shows hints of javadoc when completing a method.